crypto/modes/ccm: update api to const input buffers

cladmi opened the pull request:
### Contribution description

This changes the `cipher_encrypt_ccm` and `cipher_decrypt_ccm` functions to declare input buffers as `const`, and also the required dependencies to do it.

The tests have been updated to use `const` buffers too.


It is a API change but it is compatible as it is just adding more compatibility except if you use the function signature as the tests did.
Not sure if it requires an "API change" label in that case, I let reviewers remove it if they think it is not required.

### Testing procedure

Run the tests on a board that was working before and also compare the size difference

```
BOARD=samr21-xpro RIOT_VERSION_OVERRIDE=buildtest make  -C tests/unittests/ tests-crypto  flash test
```

The size difference I got: It removes `200` bytes ram and only adds `64` rom.

```
# master
   text    data     bss     dec     hex filename
  29292     948    2696   32936    80a8 /home/harter/work/git/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf

# const inputs
   text    data     bss     dec     hex filename
  29356     748    2696   32800    8020 /home/harter/work/git/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf
```

### Issues/PRs references

Found in https://github.com/RIOT-OS/RIOT/pull/7968#issuecomment-437962046 which is part of https://github.com/RIOT-OS/RIOT/issues/8107 resolution.

bergzand commented on sys/crypto/modes/ccm.c:
```diff
@@ -35,7 +35,7 @@ static inline int min(int a, int b)
 }
 
 int ccm_compute_cbc_mac(cipher_t* cipher, uint8_t iv[16],
```
The `iv[16]` here can also be `const`. Do you mind piggy backing that or do you think that it is unnecessary optimization here?

cladmi replied:
I only put the minimal ones that were required to make my tests pass, I did not even tried to understand :p
But sure, I will add it.

bergzand replied:
I didn't really check if it makes sense here, I just noticed that it wasn't modified by the function :)

cladmi replied:
Just to be sure, can it be an issue, in general case, for the called function that `iv` and `mac` can be the same pointer when `iv` is said const ?

Here as it is an internal function, it could even be replaced by only one "output" pointer that should contain the "iv", but that is another topic.

bergzand commented:
Ack, please squash.

While technically an API change, this allows a superset of the current functionality, so I don't consider it problematic.

cladmi commented:
Rebased on master, and squashed.

Since https://github.com/RIOT-OS/RIOT/pull/10406 was merged, the reference size was reduced a lot, but the diff stays the same.

With the PR
```
   text    data     bss     dec     hex filename
  21556     748    2696   25000    61a8 /home/harter/work/git/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf
```
In master

```
   text    data     bss     dec     hex filename
  21492     948    2696   25136    6230 /home/harter/work/git/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf
```

I could successfully execute the same test commands with `samr21-xpro` after rebase.

cladmi commented:
Thank you both for the review.
